wacz/waczindexer.py

```diff
@@ -205,12 +205,13 @@ def check_pages_and_text(self, record):
         matched_id = check_http_and_https(url, ts, self.passed_pages_dict)
         # If we find a match build a record
         if matched_id:
-            new_page = {"timestamp": ts, "url": url, "title": url}
             input_page = self.passed_pages_dict[matched_id]
+            new_page = {
+                "timestamp": ts,
+                "url": input_page.get("url") or url,
+                "title": input_page.get("title") or url,
```

Comment:
Curious what's the motivation to falling back to URL for the title if a title's not present, since we didn't before?

Reply:
It is actually doing that before, just a different way, first setting title as url and then overriding it. But agree that we should add tests for this..